jsapp/js/account/plans/plan.component.tsx

```diff
@@ -249,93 +292,76 @@ export default function Plan() {
 
         return {
           ...product,
-          prices: filteredPrices?.[0],
+          prices: filteredPrices[0],
         };
       });
 
-      return filterAmount
-        .filter((product: Product) => product.prices)
-        .sort(
-          (priceA: Price, priceB: Price) =>
-            priceA.prices.unit_amount > priceB.prices.unit_amount
-        );
+      return filterAmount.filter((price) => price.prices);
     }
     return [];
   }, [state.products, state.intervalFilter]);
 
-  const getSubscriptionsForProductId = useCallback(
-    (productId: String) =>
-      state.subscribedProduct.filter(
-        (subscription: BaseSubscription) =>
-          subscription.items[0].price.product.id === productId
-      ),
-    [state.subscribedProduct]
-  );
+  const getSubscribedProduct = useCallback(getSubscriptionsForProductId, []);
 
   const isSubscribedProduct = useCallback(
     (product: Price) => {
-      if (!product.prices.unit_amount && !hasActiveSubscription) {
+      if (!product.prices?.unit_amount && !hasActiveSubscription) {
         return true;
       }
 
-      const subscriptions = getSubscriptionsForProductId(product.id);
+      const subscriptions = getSubscribedProduct(
+        product.id,
+        state.subscribedProduct
+      );
 
-      if (subscriptions.length > 0) {
+      if (subscriptions && subscriptions.length > 0) {
```

Comment:
TypeScript doesn't like that, since we're possible comparing `undefined > 0`:
![image](https://github.com/kobotoolbox/kpi/assets/7819986/a75e1efa-5a41-4ef4-8a20-401f4cc20ff8)

Also, we need the non-null check for `subscriptions.some()` on the next line.